lib/Doctrine/ORM/Query/Filter/SQLFilter.php

```diff
@@ -72,7 +73,11 @@
     final public function setParameter($name, $value, $type = null)
     {
         if (null === $type) {
-            $type = ParameterTypeInferer::inferType($value);
+            if (is_array($type)) {
```

Comment:
Block inside this conditional is not reachable.

Reply:
Please revert this change to array inferring, the `ParameterTypeinferer` is already responsible for that itself.

Then this PR is good to go for 2.9.x in my POV.